lib/core/src/Cardano/Wallet/DB/Layer.hs

```diff
@@ -1395,194 +997,38 @@ deleteDelegationCertificates wid filters = do
-    resolveCollateralWith collateral resolvedCollateral =
-        [ (i, Map.lookup key resolvedCollateral)
-        | i <- collateral
-        , let key =
-                ( txCollateralSourceTxId i
-                , txCollateralSourceIndex i
-                )
-        ]
-
-    concatUnzip
-        :: [([a], [b], [c], [d], [e])]
-        -> (([a], [b], [c], [d], [e]))
-    concatUnzip =
-        (\(a, b, c, d, e) ->
-            ( concat a
-            , concat b
-            , concat c
-            , concat d
-            , concat e
-            )
-        ) . unzip5
-
--- | Split a query's input values into chunks, run multiple smaller queries,
--- and then concatenate the results afterwards. Used to avoid "too many SQL
--- variables" errors for "SELECT IN" queries.
-combineChunked :: [a] -> ([a] -> SqlPersistT IO [b]) -> SqlPersistT IO [b]
-combineChunked xs f = concatMapM f $ chunksOf chunkSize xs
 
+-- See also: issue #573.
 selectTxHistory
-    :: W.Wallet s
-    -> TimeInterpreter IO
+    :: Monad m
+    => W.Wallet s
+    -> TimeInterpreter m
     -> W.WalletId
     -> Maybe W.Coin
     -> W.SortOrder
-    -> [Filter TxMeta]
-    -> SqlPersistT IO [W.TransactionInfo]
-selectTxHistory cp ti wid minWithdrawal order conditions = do
-    let txMetaFilter = (TxMetaWalletId ==. wid):conditions
-    metas <- case minWithdrawal of
-        Nothing -> fmap entityVal <$> selectList txMetaFilter sortOpt
-        Just coin -> do
-            txids <- fmap (txWithdrawalTxId . entityVal)
-                <$> selectList [ TxWithdrawalAmount >=. coin ] []
-            ms <- combineChunked (nub txids) (\chunk -> selectList
-              ((TxMetaTxId <-. chunk):txMetaFilter) [])
-            let sortTxId = case order of
-                    W.Ascending -> sortOn (Down . txMetaTxId)
-                    W.Descending -> sortOn txMetaTxId
-            let sortSlot = case order of
-                    W.Ascending -> sortOn txMetaSlot
-                    W.Descending -> sortOn (Down . txMetaSlot)
-            pure $ sortSlot $ sortTxId $ fmap entityVal ms
-
-    let txids = map txMetaTxId metas
-    (ins, cins, outs, couts, ws) <- selectTxs txids
-
-    let tip = W.currentTip cp
-
-    liftIO $ txHistoryFromEntity
-        ti tip metas ins cins outs couts ws
-  where
-    -- Note: there are sorted indices on these columns.
-    -- The secondary sort by TxId is to make the ordering stable
-    -- so that testing with random data always works.
-    sortOpt = case order of
-        W.Ascending -> [Asc TxMetaSlot, Desc TxMetaTxId]
-        W.Descending -> [Desc TxMetaSlot, Asc TxMetaTxId]
-
-selectTxMeta
-    :: W.WalletId
-    -> W.Hash "Tx"
-    -> SqlPersistT IO (Maybe TxMeta)
-selectTxMeta wid tid =
-    fmap entityVal <$> selectFirst
-        [ TxMetaWalletId ==. wid, TxMetaTxId ==. (TxId tid)]
-        [ Desc TxMetaSlot ]
-
--- | Delete the transaction, but only if it's not in ledger.
--- Returns non-zero if this was a success.
```

Comment:
Perhaps the comments can be preserved — by commenting the constructors of the `DeltaTxWalletsHistory`? Even though the names give good hints, it seem sensible to me to document what the deltas are supposed to do.

Reply:
I'm not in favour of this low level of documentation, especially when the code speaks by itself
```haskell
    apply (PruneTxMetaHistory tid) (TxMetaHistory txs) =
        TxMetaHistory $ Map.alter f tid txs
      where
        f (Just tx@(TxMeta {..})) =
            if txMetaStatus == W.InLedger
                then Just tx
                else Nothing
        f Nothing = Nothing
```
And no constructors of the deltas are commented at this moment. For consistency, I should comment on everything, I guess.

Reply:
Would you agree to have a comments only PR right after this one.  Because this PR should't touch previous stores code IMO and documentation should go there

Reply:
Agreed, on both counts: The current comment does not really compress the semantics, and that it's better to do it in another PR. Thanks! 🙏